### DISCUSSION TITLE
no var/lib folder in tarball

### DISCUSSION THREAD

**Discussion by iomari** [rabbitmq/rabbitmq-server#9848] [Questions] [answered]
### Describe the bug

Failed to create PID file "/usr/local/rabbitmq_server-3.12.7/var/lib/rabbitmq/mnesia/rabbit@iomariLaptop.pid" directory: no such file or directory
Failed to create feature flags file "/usr/local/rabbitmq_server-3.12.7/var/lib/rabbitmq/mnesia/rabbit@iomariLaptop-feature_flags" directory: no such file or directory

### Reproduction steps

after instaling erlang with snap in ubuntu. 
I downloaded rabbitmq  tarball 
extrated it an moved it to /usr/local as instructed
I put /usr/local/rabbitmq_server-3.12.7/bin in the path
Then I ran rabbitmq-server and got the above errors:

### Expected behavior

the message "completed with [n] plugins.", indicating that the RabbitMQ broker has been started successfully.

**Comment by lukebakken** [comment, marked as answer]
I can't reproduce what you describe. You should provide a complete set of reproduction steps and / or transcript from your system if you would like us to continue investigating.

Here is a transcript from my Ubuntu 22.04 system:

```
lbakken@PROKOFIEV ~/issues/rabbitmq/rabbitmq-server/gh-9847
$ curl -LO https://github.com/rabbitmq/rabbitmq-server/releases/download/v3.12.8/rabbitmq-server-generic-unix-3.12.8.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 14.9M  100 14.9M    0     0  15.0M      0 --:--:-- --:--:-- --:--:-- 15.0M

lbakken@PROKOFIEV ~/issues/rabbitmq/rabbitmq-server/gh-9847
$ tar xf rabbitmq-server-generic-unix-3.12.8.tar.xz

lbakken@PROKOFIEV ~/issues/rabbitmq/rabbitmq-server/gh-9847
$ sudo mv rabbitmq_server-3.12.8 /usr/local
[sudo] password for lbakken:

lbakken@PROKOFIEV ~/issues/rabbitmq/rabbitmq-server/gh-9847
$ export PATH="/usr/local/rabbitmq_server-3.12.8/sbin:$PATH"

lbakken@PROKOFIEV ~/issues/rabbitmq/rabbitmq-server/gh-9847
$ rabbitmq-server
2023-11-01 12:10:35.623504-07:00 [notice] <0.44.0> Application syslog exited with reason: stopped
2023-11-01 12:10:35.627886-07:00 [notice] <0.235.0> Logging: switching to configured handler(s); following messages may not be visible in this log output

  ##  ##      RabbitMQ 3.12.8
  ##  ##
  ##########  Copyright (c) 2007-2023 VMware, Inc. or its affiliates.
  ######  ##
  ##########  Licensed under the MPL 2.0. Website: https://rabbitmq.com

  Erlang:      26.1.2 [jit]
  TLS Library: OpenSSL - OpenSSL 3.1.3 19 Sep 2023
  Release series support status: supported

  Doc guides:  https://rabbitmq.com/documentation.html
  Support:     https://rabbitmq.com/contact.html
  Tutorials:   https://rabbitmq.com/getstarted.html
  Monitoring:  https://rabbitmq.com/monitoring.html

  Logs: /usr/local/rabbitmq_server-3.12.8/var/log/rabbitmq/rabbit@PROKOFIEV.log
        <stdout>

  Config file(s): (none)

  Starting broker... completed with 0 plugins.
```

**Comment by michaelklishin** [comment]
RabbitMQ generally will not create its own **top level** data directory. If you want to use the generic binary build, it is **up to you** to do that.

Ubuntu users should follow [our Debian and Ubuntu installation guide](https://rabbitmq.com/install-debian.html) and instead of using an Erlang snap + the binary build, use our Debian package repositories. The Debian package does set up a data directory and other essentials of this kind.

And more importantly, they provide the latest and greatest Erlang, including for aarch64 (via Launchpad), while standard packages and snaps on Ubuntu are usually a year or two behind. For example, the Erlang snap is maintained by a well known Erlang community member, and even then it only provides 25.3 in its "stable channel". So it won't be an option for running RabbitMQ 3.13 which should ship in Nov-Dec 2023.
